### PR TITLE
Nicolas.laurenti/data mapper

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
In this PR, I map the app data

**_Motorola Z Play and Folders :_**

ezgif com-gif-maker (3) image

### CODE REVIEW

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Data.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.onboarding.wheatherapp.data.entity
+
+
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/entity/Data.kt]: remove this extra line pls

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Main.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.onboarding.wheatherapp.data.entity
+
+data class Main(
+     val temp: String,
+     val tempMin: String,
+     val tempMax: String
+)
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/entity/Main.kt]: add empty line at the end of the file pls

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Weather.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.onboarding.wheatherapp.data.entity
+
+data class Weather(
+    val list: ArrayList<Data>
+)
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/entity/Weather.kt]: add empty line at the end of the file pls

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        val listData: ArrayList<Data> = arrayListOf()
+        list.forEach {
+            listData.add(Data(transformMain(it.main)))
+        }
+        return listData
+    }
+
+    fun transformWeather(weatherResponse: WeatherResponse) = Weather(transformList(weatherResponse.list))
+
+}
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: add empty line at the end of the file pls

File: app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt [hunk 5]
```diff
@@ -1,21 +1,23 @@
...
+package com.onboarding.wheatherapp.data.services
 
-import com.onboarding.wheatherapp.services.api.WeatherApi
-import com.onboarding.wheatherapp.services.request.generator.WeatherRequestGenerator
-import com.onboarding.wheatherapp.services.response.WeatherResponse
+import com.onboarding.wheatherapp.data.entity.Weather
+import com.onboarding.wheatherapp.data.mapper.WeatherMapper
+import com.onboarding.wheatherapp.data.services.api.WeatherApi
+import com.onboarding.wheatherapp.data.services.request.generator.WeatherRequestGenerator
 import io.reactivex.rxjava3.core.Observable
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt]: remove this extra line pls

**Comment by BQuinteros** [comment]
Why you put a screen of the logcat? It does not give useful information

File: app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt (hunk 5 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Weather.kt (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/entity/Weather.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Main.kt (hunk 2 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/entity/Main.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/data/entity/Data.kt (hunk 1 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/entity/Data.kt]: Solved!

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**